darts/dataprocessing/transformers/boxcox.py

```diff
@@ -61,35 +75,92 @@ def fit(self,
+                         logger)
+            else:
+                # Replicate lmbda to match dimensions of the time series
+                lmbda = [lmbda] * data.width
+
+            return lmbda
+
+        if isinstance(data, TimeSeries):
+            self._lmbda = _fit_single_series(data, self._lmbda_original)
+
+        elif isinstance(data, Sequence):
+
+            # in case of sequence of sequence of values, the outer sequence must be as long as the number of time series
+            if isinstance(self._lmbda_original, Sequence) and isinstance(self._lmbda_original[0], Sequence):
+                raise_if(len(self._lmbda_original) != len(data),
+                         "with multiple time series the number of lmbdas sequences must equal the number of time \
+                         series",
+                         logger)
+            else:
+                # we have either a single value or None
+                iterator = _build_tqdm_iterator(data,
+                                                verbose=self._verbose,
+                                                desc="Fitting BoxCox")
+
+                self._lmbda = Parallel(n_jobs=self._n_jobs)(delayed(_fit_single_series)
+                                                            (ts, self._lmbda_original)
+                                                            for ts in iterator)
 
         return self
 
-    def transform(self, data: TimeSeries, *args, **kwargs) -> TimeSeries:
+    def transform(self,
+                  data: Union[TimeSeries, Sequence[TimeSeries]],
+                  *args,
+                  **kwargs) -> Union[TimeSeries, Sequence[TimeSeries]]:
         super().transform(data, *args, **kwargs)
 
-        def _boxcox_wrapper(col):
-            idx = data._df.columns.get_loc(col.name)  # get index from col name
-            return boxcox(col, self._lmbda[idx])
+        def _transform_single_series(data, lmbda):
+            def _boxcox_wrapper(col):
+                idx = data._df.columns.get_loc(col.name)  # get index from col name
+                return boxcox(col, lmbda[idx])
 
-        return TimeSeries.from_dataframe(data._df.apply(_boxcox_wrapper))
+            return TimeSeries.from_dataframe(data._df.apply(_boxcox_wrapper))
 
-    def inverse_transform(self, data: TimeSeries, *args, **kwargs) -> TimeSeries:
+        if isinstance(data, TimeSeries):
+            return _transform_single_series(data, self._lmbda)
+        else:
+            # multiple time series
+            iterator = _build_tqdm_iterator(zip(data, self._lmbda),
+                                            verbose=self._verbose,
+                                            desc="Transforming BoxCox",
+                                            total=len(data))
+
+            transformed_data = Parallel(n_jobs=self._n_jobs)(delayed(_transform_single_series)(ts, lmbda)
```

Comment:
Are you sure `transformed_data` is always of type `Sequence` ? If yes can it be any type of `Sequence` ?

Reply:
`transformed_data` is always a `Sequence`, specifically, a `List`. I modified the types accordingly.